File: examples/java7_tienda/TiendaManager.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Date;
import java.text.SimpleDateFormat;

/**
 * Sistema de Gestión de Tienda - Java 7
 * Código legacy que usa patrones antiguos:
 * - Sin lambdas, sin streams
 * - Raw types en colecciones
 * - Manejo manual de iteradores
 * - Sin try-with-resources
 */
public class TiendaManager {

    private ArrayList productos = new ArrayList();
    private HashMap inventario = new HashMap();
    private ArrayList ventas = new ArrayList();
    private double totalVentas = 0.0;

    public static void main(String[] args) {
        TiendaManager tienda = new TiendaManager();
        tienda.inicializar();
        tienda.registrarVentas();
        tienda.mostrarReporte();
    }

    public void inicializar() {
        System.out.println("==============================================");
        System.out.println("  SISTEMA DE GESTION DE TIENDA v1.0");
        System.out.println("  Java 7 - Legacy System");
        System.out.println("==============================================");

        // Agregar productos (sin diamond operator en Java 7 style)
        HashMap prod1 = new HashMap();
        prod1.put("codigo", "SKU001");
        prod1.put("nombre", "Arroz Extra 5kg");
        prod1.put("precio", "18.50");
        prod1.put("categoria", "Abarrotes");
        productos.add(prod1);

        HashMap prod2 = new HashMap();
        prod2.put("codigo", "SKU002");
        prod2.put("nombre", "Aceite Vegetal 1L");
        prod2.put("precio", "12.90");
        prod2.put("categoria", "Abarrotes");
        productos.add(prod2);

        HashMap prod3 = new HashMap();
        prod3.put("codigo", "SKU003");
        prod3.put("nombre", "Detergente Bolsa 2kg");
        prod3.put("precio", "24.00");
        prod3.put("categoria", "Limpieza");
        productos.add(prod3);

        HashMap prod4 = new HashMap();
        prod4.put("codigo", "SKU004");
        prod4.put("nombre", "Leche Evaporada x6");
        prod4.put("precio", "22.80");
        prod4.put("categoria", "Lacteos");
        productos.add(prod4);

        // Inicializar inventario
        inventario.put("SKU001", new Integer(150));
        inventario.put("SKU002", new Integer(200));
        inventario.put("SKU003", new Integer(80));
        inventario.put("SKU004", new Integer(120));

        System.out.println("  Productos cargados: " + productos.size());
        System.out.println("  Inventario inicializado");
        System.out.println();
    }

    public void registrarVentas() {
        System.out.println("--- REGISTRANDO VENTAS DEL DIA ---");

        registrarVenta("SKU001", 3);
        registrarVenta("SKU002", 5);
        registrarVenta("SKU003", 2);
        registrarVenta("SKU001", 10);
        registrarVenta("SKU004", 4);
        registrarVenta("SKU002", 8);

        System.out.println();
    }

    private void registrarVenta(String codigo, int cantidad) {
        // Buscar producto con iterador (Java 7 style)
        Iterator it = productos.iterator();
        HashMap productoEncontrado = null;

        while (it.hasNext()) {
            HashMap prod = (HashMap) it.next();
            if (prod.get("codigo").equals(codigo)) {
                productoEncontrado = prod;
                break;
            }
        }

        if (productoEncontrado != null) {
            double precio = Double.parseDouble((String) productoEncontrado.get("precio"));
            double subtotal = precio * cantidad;
            totalVentas += subtotal;

            // Actualizar inventario
            Integer stock = (Integer) inventario.get(codigo);
            if (stock != null) {
                inventario.put(codigo, new Integer(stock.intValue() - cantidad));
            }

            // Registrar venta
            HashMap venta = new HashMap();
            venta.put("codigo", codigo);
            venta.put("producto", productoEncontrado.get("nombre"));
            venta.put("cantidad", String.valueOf(cantidad));
            venta.put("subtotal", String.valueOf(subtotal));
            venta.put("fecha", new SimpleDateFormat("yyyy-MM-dd HH:mm").format(new Date()));
            ventas.add(venta);

            System.out.println("  + Venta: " + cantidad + "x " + productoEncontrado.get("nombre") + " = S/." + String.format("%.2f", subtotal));
        }
    }

    public void mostrarReporte() {
        System.out.println("==============================================");
        System.out.println("  REPORTE DE VENTAS");
        System.out.println("==============================================");
        System.out.println("  Total ventas del dia: S/." + String.format("%.2f", totalVentas));
        System.out.println("  Transacciones: " + ventas.size());
        System.out.println();

        // Alertas de stock bajo
        System.out.println("--- ALERTAS DE INVENTARIO ---");
        Iterator entries = inventario.entrySet().iterator();
        int alertas = 0;
        while (entries.hasNext()) {
            HashMap.Entry entry = (HashMap.Entry) entries.next();
            Integer stock = (Integer) entry.getValue();
            if (stock.intValue() < 100) {
                alertas++;
                System.out.println("  ! STOCK BAJO: " + entry.getKey() + " (quedan: " + stock + " unidades)");
            }
        }

        if (alertas == 0) {
            System.out.println("  Todos los productos con stock suficiente");
        }

        System.out.println();
        System.out.println("==============================================");
        System.out.println("  Fin del reporte - " + new SimpleDateFormat("yyyy-MM-dd").format(new Date()));
        System.out.println("==============================================");
    }
}

File: output/ai_java7_to_python/TiendaManager.py
from dataclasses import dataclass
from datetime import datetime
from pathlib import Path
from typing import List, Dict

@dataclass
class Producto:
    codigo: str
    nombre: str
    precio: float
    categoria: str

class TiendaManager:
    def __init__(self):
        self.productos: List[Producto] = []
        self.inventario: Dict[str, int] = {}
        self.ventas: List[Dict] = []
        self.total_ventas: float = 0.0

    def inicializar(self):
        print("==============================================")
        print("  SISTEMA DE GESTION DE TIENDA v1.0")
        print("  Python 3.12")
        print("==============================================")

        productos_data = [
            {"codigo": "SKU001", "nombre": "Arroz Extra 5kg", "precio": 18.50, "categoria": "Abarrotes"},
            {"codigo": "SKU002", "nombre": "Aceite Vegetal 1L", "precio": 12.90, "categoria": "Abarrotes"},
            {"codigo": "SKU003", "nombre": "Detergente Bolsa 2kg", "precio": 24.00, "categoria": "Limpieza"},
            {"codigo": "SKU004", "nombre": "Leche Evaporada x6", "precio": 22.80, "categoria": "Lacteos"},
        ]

        for prod_data in productos_data:
            self.productos.append(Producto(**prod_data))

        self.inventario = {
            "SKU001": 150,
            "SKU002": 200,
            "SKU003": 80,
            "SKU004": 120,
        }

        print(f"  Productos cargados: {len(self.productos)}")
        print("  Inventario inicializado")
        print()

    def registrar_ventas(self):
        print("--- REGISTRANDO VENTAS DEL DIA ---")

        ventas_data = [
            ("SKU001", 3),
            ("SKU002", 5),
            ("SKU003", 2),
            ("SKU001", 10),
            ("SKU004", 4),
            ("SKU002", 8),
        ]

        for codigo, cantidad in ventas_data:
            self.registrar_venta(codigo, cantidad)

        print()

    def registrar_venta(self, codigo: str, cantidad: int):
        producto = next((p for p in self.productos if p.codigo == codigo), None)
        if producto:
            subtotal = producto.precio * cantidad
            self.total_ventas += subtotal

            self.inventario[codigo] -= cantidad

            venta = {
                "codigo": codigo,
                "producto": producto.nombre,
                "cantidad": cantidad,
                "subtotal": subtotal,
                "fecha": datetime.now().strftime("%Y-%m-%d %H:%M"),
            }
            self.ventas.append(venta)

            print(f"  + Venta: {cantidad}x {producto.nombre} = S/. {subtotal:.2f}")

    def mostrar_reporte(self):
        print("==============================================")
        print("  REPORTE DE VENTAS")
        print("==============================================")
        print(f"  Total ventas del dia: S/. {self.total_ventas:.2f}")
        print(f"  Transacciones: {len(self.ventas)}")
        print()

        print("--- ALERTAS DE INVENTARIO ---")
        alertas = [f"  ! STOCK BAJO: {codigo} (quedan: {stock} unidades)" 
                   for codigo, stock in self.inventario.items() if stock < 100]

        if alertas:
            for alerta in alertas:
                print(alerta)
        else:
            print("  Todos los productos con stock suficiente")

        print()
        print("==============================================")
        print(f"  Fin del reporte - {datetime.now().strftime('%Y-%m-%d')}")
        print("==============================================")

def main():
    tienda = TiendaManager()
    tienda.inicializar()
    tienda.registrar_ventas()
    tienda.mostrar_reporte()

if __name__ == "__main__":
    main()